dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginTest.kt

```diff
@@ -360,6 +375,304 @@ internal class DdAndroidGradlePluginTest {
+        val variantName = "$flavorName${buildTypeName.capitalize()}"
+        whenever(mockVariant.name) doReturn variantName
+        whenever(mockVariant.flavorName) doReturn flavorName
+        whenever(mockVariant.versionName) doReturn versionName
+        whenever(mockVariant.applicationId) doReturn packageName
+        whenever(mockVariant.buildType) doReturn mockBuildType
+
+        val fakeCompileTask = fakeProject.task("compile${variantName.capitalize()}Sources")
+
+        val mockConfiguration = mock<Configuration>()
+        val mockResolvedConfiguration = mock<ResolvedConfiguration>()
+
+        whenever(mockResolvedConfiguration.firstLevelModuleDependencies) doReturn emptySet()
+        whenever(mockConfiguration.resolvedConfiguration) doReturn mockResolvedConfiguration
+        whenever(mockVariant.runtimeConfiguration) doReturn mockConfiguration
+
+        // When + Then
+        val exception = assertThrows<MissingSdkException> {
+            testedPlugin.configureVariantForSdkCheck(
+                fakeProject,
+                mockVariant,
+                fakeExtension
+            )!!.actions.first().execute(fakeCompileTask)
+        }
+
+        assertThat(exception.message)
+            .isEqualTo(DdAndroidGradlePlugin.MISSING_DD_SDK_MESSAGE.format(variantName))
+    }
+
+    @Test
+    fun `𝕄 no exception thrown 𝕎 configureVariantForSdkCheck() { sdk is missing w warning set }`(
+        @StringForgery(case = Case.LOWER) flavorName: String,
+        @StringForgery(case = Case.LOWER) buildTypeName: String,
+        @StringForgery versionName: String,
+        @StringForgery packageName: String
+    ) {
+        // Given
+        fakeExtension.checkProjectDependencies = SdkCheckLevel.WARN
+        val variantName = "$flavorName${buildTypeName.capitalize()}"
+        whenever(mockVariant.name) doReturn variantName
+        whenever(mockVariant.flavorName) doReturn flavorName
+        whenever(mockVariant.versionName) doReturn versionName
+        whenever(mockVariant.applicationId) doReturn packageName
+        whenever(mockVariant.buildType) doReturn mockBuildType
+
+        val fakeCompileTask = fakeProject.task("compile${variantName.capitalize()}Sources")
+
+        val mockConfiguration = mock<Configuration>()
+        val mockResolvedConfiguration = mock<ResolvedConfiguration>()
+
+        whenever(mockResolvedConfiguration.firstLevelModuleDependencies) doReturn emptySet()
+        whenever(mockConfiguration.resolvedConfiguration) doReturn mockResolvedConfiguration
+        whenever(mockVariant.runtimeConfiguration) doReturn mockConfiguration
+
+        // When + Then
+        assertDoesNotThrow {
+            testedPlugin.configureVariantForSdkCheck(
+                fakeProject,
+                mockVariant,
+                fakeExtension
+            )!!.actions.first().execute(fakeCompileTask)
+        }
+    }
+
+    @Test
+    fun `𝕄 do nothing 𝕎 configureVariantForSdkCheck() { sdk is there }`(
```

Comment:
You're missing a `𝕄 do nothing 𝕎 configureVariantForSdkCheck() { sdk is missing w check = none }`

Reply:
indeed, good catch!